src/lib/MrDocsCompilationDatabase.cpp

```diff
@@ -507,18 +514,40 @@ adjustCommandLine(
     while (idx < cmdline.size())
     {
         // Parse one argument as a Clang option
-        // ParseOneArg updates Index to the next argument to be parsed.
-        unsigned const idx0 = idx;
         std::unique_ptr<llvm::opt::Arg> arg =
             opts_table.ParseOneArg(args, idx, visibility);
         if (!isValidMrDocsOption(workingDir, arg))
         {
             continue;
         }
-        new_cmdline.insert(
-            new_cmdline.end(),
-            cmdline.begin() + idx0,
-            cmdline.begin() + idx);
+
+        // Translate clang-cl /std: into clang -std=
+        const llvm::opt::Option opt =
+            arg->getOption().getUnaliasedOption();
+        if (opt.matches(clang::driver::options::OPT__SLASH_std))
+        {
+            MRDOCS_ASSERT(arg->getNumValues() == 1);
+
+            std::string_view v = arg->getValue();
+            if (v == "c++latest" || v == "c++23preview")
```

Comment:
This is cumbersome. Since we have access to the whole Clang library, I'm sure there should be some way to know what the highest std is. Even if it's some enum of standards somewhere, we can evaluate an inline constexpr requires to check if STD_NUMBER exists in that enum.

A few ways to get the highest stardard could be:

```
#include "clang/Frontend/LangStandards.def"
```
 
or

```
clang::getLangStandardForName("c++23")
```

or

```
bool ok = clang::CompilerInvocation::CreateFromArgs(
    CI,
    {"-std=c++23"},
    *Diags
);
```

there must be any other ways.

These functions only give you if one specific standard is there, but you can just iterate 23 + X * 3 until you find the highest valid number, potentially at compile time.